8.x/theme-development/helpers.md

```diff
@@ -1,100 +1,144 @@
-```
-This function unparses the URL. As you know, there is a `pars_url` function in PHP, which works as follows:
+To test these functions, add a new blade file to the `larammerce-base-them` directory, then follow these steps:
+
+
+1. Create a directory in the project root from the admin panel.
+2. Click the plus button located at the bottom left of the page.
+3. Enter a title and URL, and enable the "has webpage" option in the form.
+- Note: the URl name should be same as the blade file name.
+4. Save the changes.
+5. After saving the changes, a new form will display. Choose "editing webpage" from the dropdown menu and select the created blade file name.
+6. Save and exit.
+7. Write your desired functions (they are mentioned as `Example` in this document) in the blade file, run `./deploy.sh` in ypur terminal in the `larammerce-base-them` directory and view their results on the specified URL.
+
+![create webpage in admin panel](/helpers/CreatePage.png)
+
+---
+
+### Helper functions description
+
+#### parse_url 
+
+*This is a php function which parses a URL and returns an associative array containing any of the various components of the URL that are present(such as `scheme`, `user`. `pass`, `host`, `port`,`path`, `query`, `fragment`). The values of the array elements are not URL decoded.* *<sup>[1](#1)</sup>*
+
+To do so, lets define a url and parse it:
 
 ```php
 @php
 
 $url= "https://google.com/salam/donya?test=sample";
-parse_url($url);
+dd(parse_url($url));
 
 @endphp
 ```
-Output
-``` 
+in `larammerce-base-theme` directory run:
+
+```bash
+./deploy.sh
+```
+
+Output:
+```bash
 array:4 [▼
   "scheme" => "https"
   "host" => "google.com"
   "path" => "/salam/donya"
   "query" => "test=sample"
 ]
 ```
-Each part of the URL that is ‍‍‍`scheme`, `host`, `path` and  `query` is displayed.
+As you see URL components are returned as an array.
 
+---
 
-Now an example is given to check the `unpars_url` function. Empty the `query` and unparse the URL.
+#### unparse_url
 
 ```php
-@php
+function unparse_url(array $parsed_url, array $ommit = []): string {
+    ...
+}
+```
+This function takes the parsed URL array as input, merges all the components, and as the result, returns a unified URL. 
+
+For instance, let's change the `query's` value and unparse the URL and see the result:
 
+```php{5,6}
+@php
+ 
 $url= "https://google.com/salam/donya?test=sample";
 $parsed_url = parse_url($url);
-
 $parsed_url["query"] = " ";
 $new_url = unparse_url($parsed_url);
 
 @endphp
 
 <p> the changed url : {{ $new_url }}</p>
 ```
-Output
-``` 
-"https://google.com/salam/donya? "
-```
-
-It can be seen that the query is removed from the URL.
+Output:
 
-This function takes an array of parsed URLs from the input and merges these into `scheme`, `user`, `pass`, `host`, `port`, `path`, `query`, `fragment` in the output and transforms them into a unified URL.
-___________
+```html
+<p> "https://google.com/salam/donya? " </p>
```

Comment:
<p> "https://google.com/salam/donya? " </p>  ->  the changed url: https://google.com/salam/donya?